### PR TITLE
[Ace] Support -fPIC compiler flag on Linux

### PR DESCRIPTION
**Pull request by pierrebizz**
Add fpic feature to enable the -fPIC flag in order to be able to link ACE/TAO in a dynamic library

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <linux>, <No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

### CODE REVIEW

**Comment by Hoikas** [comment]
My understanding is that vcpkg prefers to always pass `-fPIC`, so this should not be a feature, rather just a bugfix.

**Comment by cenit** [comment]
fpic is not a feature imho, it should be enabled by default, always!

**Comment by cenit** [comment]
> My understanding is that vcpkg prefers to always pass `-fPIC`, so this should not be a feature, rather just a bugfix.

yes, i think this is the rule here. Please remove the feature and apply the flag **always**

**Comment by pierrebizz** [comment]
Very well; is there a risk that it will break the builds of people already using this config without the fPIC flag ?

**Comment by Neumann-A** [comment]
This is a case for `z_vcpkg_get_cmake_vars` and not manually adding flags.

**Comment by pierrebizz** [comment]
Can someone more intelligent than me explain what exactly failed in the x64_windows_static build ?

**Comment by cenit** [comment]
```
Error: vcpkg has crashed; no additional details are available.
The source line is D:\a\_work\1\s\src\vcpkg\binarycaching.cpp(554)
```

not related to your PR.
vcpkg ci is somehow broken, HTTP remotes are pointing to non-existing address (err 403)

**Comment by pierrebizz** [comment]
Do I have to do something to retrigger ?

**Comment by cenit** [comment]
you can merge with master if your branch is not already fully up to date, or you can just wait for a maintainer with the power to launch an /azp run command

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -105,7 +105,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
 elseif(VCPKG_TARGET_IS_LINUX)
   set(SOLUTION_TYPE gnuace)
   set(config_h_contents "#include \"ace/config-linux.h\"\n")
-  file(WRITE "${ACE_ROOT}/include/makeinclude/platform_macros.GNU" "include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU")
+  file(WRITE "${ACE_ROOT}/include/makeinclude/platform_macros.GNU" "CCFLAGS += -fPIC\ninclude $(ACE_ROOT)/include/makeinclude/platform_linux.GNU")
```
Comment (jwillemsen) [inline ports/ace/portfile.cmake:108]: At the moment you build ACE/TAO as shared library we do add `-fPIC` already to the compiler flags, no need to do that manually. As far as I know vcpkg does a static build, at that moment `-fPIC` is not used

Comment (jwillemsen) [reply ports/ace/portfile.cmake:108]: Ok, I read in the discussion that vcpkg always wants to use `-fPIC`

**Comment by Cheney-W** [comment]
@pierrebizz Could you please solve the conflict?

**Comment by ras0219-msft** [comment]
@Neumann-A's point is correct, the best long term solution is to use `vcpkg_cmake_get_vars()` from the `vcpkg-cmake` helper and not to hardcode flags.

However, without someone signed up to do that work, I don't want to block this reasonable incremental improvement. @pierrebizz I'd encourage you to consider submitting a PR that uses the above function to get the full set of CMake settings, which would _greatly_ improve compatibility in more ways than just `-fPIC`.

Thanks again!

**Pull request merged**